servers/audio/effects/audio_effect_sample_rate.cpp

```diff
@@ -0,0 +1,86 @@
+/*************************************************************************/
+/*  audio_effect_sample_rate.cpp                                         */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2022 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2022 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "audio_effect_sample_rate.h"
+#include "core/math/math_funcs.h"
+#include "servers/audio_server.h"
+
+void AudioEffectSampleRateInstance::process(const AudioFrame *p_src_frames, AudioFrame *p_dst_frames, int p_frame_count) {
+	float frames_until_next_sample = AudioServer::get_singleton()->get_mix_rate() / base->rate;
+
+	for (int i = 0; i < p_frame_count; i++) {
+		if (processed_frames >= frames_until_next_sample) {
+			last_sampled_frame = p_src_frames[i]; // sample a new frame
+			processed_frames = Math::fmod(processed_frames - frames_until_next_sample, 1.0f);
+		}
+
+		// output dry/wet signal based on the `mix` control
```

Comment:
```
// Output dry/wet signal based on the `mix` control.
```
We've recently standardized on capitalized / full stops for new comments, so I suppose we should be pedantic. :grin: